tslearn/matrix_profile.py

```diff
@@ -91,15 +100,22 @@ class MatrixProfile(TransformerMixin,
 
     References
     ----------
-    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al. 
-       Matrix Profile I: All Pairs Similarity Joins for Time Series: A 
+    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al.
+       Matrix Profile I: All Pairs Similarity Joins for Time Series: A
        Unifying View that Includes Motifs, Discords and Shapelets.
        ICDM 2016.
+
+    .. [2] STUMPY documentation https://stumpy.readthedocs.io/en/latest/
+
     """
 
-    def __init__(self, subsequence_length=1, scale=True):
+    def __init__(
+        self, subsequence_length=1, implementation="numpy", scale=True
+    ):
         self.subsequence_length = subsequence_length
         self.scale = scale
+        self.implementation = implementation
+        self._available_implementations = ["numpy", "stump", "gpu_stump"]
```

Comment:
you should not set such parameter in `__init__` (sklearn conventions). And will you use it in several places or just one? Another option would be to set it as a module-level variable

Reply:
Just in one place, thought it would be easier to add things to add afterwards. Will move it to the appropriate place.

Reply:
By the way, why do you use "stump" and "gpu_stump" instead of "stumpy" that more clearly relates to the so-called library?

Reply:
I just took the names of the implementations from the STUMPY library itself. Do you think it would be less confusing to use "stumpy" and "gpu_stumpy", even though STUMPY methods are `stumpy.stump()` and `stumpy.gpu_stump()` ?